web-app/src/lib/db/repositories/job-schedule.repository.ts

```diff
@@ -0,0 +1,158 @@
+    const maybeEndAfter = (
+      data as unknown as { endAfterOccurrences?: number | null }
+    ).endAfterOccurrences;
+    if (maybeEndAfter !== undefined)
+      payloadObj.endAfterOccurrences = maybeEndAfter;
+
+    return await tx.jobSchedule.create({
+      data: payloadObj as unknown as Prisma.JobScheduleCreateInput,
+    });
+  },
+
+  async update(
+    id: string,
+    data: Prisma.JobScheduleUpdateInput,
+    tx: Prisma.TransactionClient = prisma,
+  ) {
+    return await tx.jobSchedule.update({ where: { id }, data });
+  },
+
+  async delete(id: string, tx: Prisma.TransactionClient = prisma) {
+    return await tx.jobSchedule.delete({ where: { id } });
+  },
+
+  async getById(id: string, tx: Prisma.TransactionClient = prisma) {
+    return await tx.jobSchedule.findUnique({ where: { id } });
+  },
+
+  async findDue(limit: number = 50, tx: Prisma.TransactionClient = prisma) {
+    return await tx.jobSchedule.findMany({
+      where: {
+        isActive: true,
+        nextRunAt: { lte: new Date() },
+      },
+      orderBy: { nextRunAt: "asc" },
+      take: limit,
+    });
+  },
+
+  async getScheduleJobsByAgentIdAndContext(
+    agentId: string,
+    userId: string,
+    organizationId: string | null,
+    tx: Prisma.TransactionClient = prisma,
+  ) {
+    const organizationFilter: Prisma.JobScheduleWhereInput[] = organizationId
+      ? [{ organizationId }, { organizationId: null }]
+      : [{ organizationId: null }];
+
+    return await tx.jobSchedule.findMany({
+      where: {
+        agentId,
+        userId,
+        OR: organizationFilter,
+      },
+      orderBy: { createdAt: "desc" },
+      include: {
+        agent: true,
+      },
+    });
+  },
+
+  async getScheduleJobsByContext(
+    userId: string,
+    organizationId: string | null,
+    tx: Prisma.TransactionClient = prisma,
+  ) {
+    const organizationFilter: Prisma.JobScheduleWhereInput[] = organizationId
+      ? [{ organizationId }, { organizationId: null }]
+      : [{ organizationId: null }];
+
+    return await tx.jobSchedule.findMany({
+      where: {
+        userId,
+        AND: organizationFilter,
+      },
+      orderBy: { updatedAt: "desc" },
+      include: {
+        agent: true,
+      },
+    });
+  },
+
+  async markRunAttempt(id: string, tx: Prisma.TransactionClient = prisma) {
+    return await tx.jobSchedule.update({
+      where: { id },
+      data: {
+        lastRunAt: new Date(),
+        // increment requires field to exist; cast to allow increment syntax
+        ...({
+          occurrenceCount: { increment: 1 },
+        } as unknown as Prisma.JobScheduleUpdateInput),
+      },
+    });
+  },
+
+  async setNextRun(
+    id: string,
+    nextRunAt: Date | null,
+    tx: Prisma.TransactionClient = prisma,
+  ) {
+    return await tx.jobSchedule.update({
+      where: { id },
+      data: {
+        nextRunAt,
+        isActive: nextRunAt ? true : false,
+        pauseReason: null,
+      },
+    });
+  },
+
+  async setPaused(
```

Comment:
I would recommend to call this `setActive`, because you can do both pausing and resuming.